bug: stop Codex compact probe from selecting memory requests

## Background

PR #28121 updated the runner's Codex compact-generation compatibility probe to use Codex app-server 0.148.0 with the production `features.memories` setting enabled.

The `runner-behavior-lane-b` check now fails intermittently in unrelated pull requests. One example is https://github.com/vm0-ai/vm0/actions/runs/32230179810/job/95999170150?pr=28157.

## Problem and evidence

The loopback HTTP handler records every `/v1/responses` request, while `resume_history` assumes the last recorded request belongs to the foreground compatibility turn:

```python
return resumed, history_file, original_size, requests[-1][1]
```

With memories enabled, Codex can issue a background memory-generation request against the same loopback endpoint. Its completion order relative to the foreground turn is nondeterministic. When the memory request is last, the probe compares or inspects the wrong payload and reports one of these errors:

```text
AssertionError: Codex reconstructed different model input from the compact generation
AssertionError: Codex did not reconstruct compact replacement history
```

In a sample of the latest 11 completed runs that actually executed `runner-behavior-lane-b`, 6 failed and 5 passed. Local instrumentation also captured both payloads: the intended request used `gpt-5.1-codex-mini` and ended with `CODEX-COMPACT-APPEND-TOKEN`, while the unrelated memory request used a different model and developer input.

This is a probe request-correlation race, not a compact-history reconstruction failure. The exit code 137 seen during cleanup is a consequence of cancelling the long-running test job after the assertion fails.

## Decision

Correlate captured `/v1/responses` payloads with the foreground turn's unique append prompt and require exactly one match. Use that correlated payload for both plain JSONL resume comparison and zstd materialization coverage.

Keep `features.memories = true`. Background memory traffic is valid production behavior and should remain present without affecting which foreground request the probe inspects.

## Alternatives considered

- **Keep selecting the last request and retry failures:** rejected because request ordering is inherently nondeterministic and retries would only mask the race.
- **Disable memories in the probe:** rejected because it would stop mirroring the production app-server configuration and hide valid concurrent traffic.
- **Select by model name:** rejected because model selection is configuration-dependent and is a weaker correlation key than the unique prompt owned by this probe.

## Risks and mitigations

- A future wire-format change could alter the terminal prompt message shape. This is acceptable compatibility-test sensitivity; the exact-one-match assertion will fail explicitly instead of silently checking an unrelated request.
- A background request could unexpectedly contain the unique token. Requiring exactly one match turns that ambiguity into a deterministic diagnostic rather than an order-dependent result.
- The change affects CI probe selection only; it does not alter runner, guest-agent, persisted history, or deployment behavior.

## Acceptance criteria

- Select the foreground `/v1/responses` payload by the probe's unique append prompt instead of request arrival order.
- Assert that exactly one matching foreground request was captured.
- Preserve `features.memories = true` so the probe continues to mirror production app-server configuration.
- Apply the same request-correlation rule to both plain JSONL resume and zstd materialization coverage.
- Keep the existing compact-history, thread-ID, append, and materialization assertions.
- Do not add retries, sleeps, or skipped coverage.
- Repeated executions of the Codex compact-generation probe pass with Codex CLI 0.148.0, including executions where a background memory request is captured.

## Relevant code

- `.github/scripts/runner-behavior-exec.sh`
- PR #28121

--- a/.github/scripts/runner-behavior-exec.sh
+++ b/.github/scripts/runner-behavior-exec.sh
@@ -1352,6 +1352,22 @@ def app_server_output(result):
     return json.dumps(result["messages"]) + "\n" + result["stderr"]
 
 
+def response_request_for_prompt(prompt):
+    prompt_content = [{"type": "input_text", "text": prompt}]
+    matching_requests = [
+        payload
+        for path, payload in requests
+        if path == "/v1/responses"
+        and payload["input"][-1].get("role") == "user"
+        and payload["input"][-1].get("content") == prompt_content
+    ]
+    assert len(matching_requests) == 1, (
+        "expected exactly one Responses request for the probe prompt, "
+        f"found {len(matching_requests)} among {len(requests)} requests"
+    )
+    return matching_requests[0]
+
+
 def zstd_compress(history_bytes):
     result = subprocess.run(
         [
@@ -1428,10 +1444,8 @@ def resume_history(
     assert (
         f"no rollout found for thread id {session_id}" not in output.lower()
     ), output
-    assert requests and requests[-1][0] == "/v1/responses", (
-        f"Codex did not reach the loopback Responses endpoint\n{output}"
-    )
-    return resumed, history_file, original_size, requests[-1][1]
+    response_request = response_request_for_prompt(append_token)
+    return resumed, history_file, original_size, response_request
 
 
 def probe_current_codex(
@@ -1529,9 +1543,7 @@ def probe_zstd_materialization(
     assert (
         f"no rollout found for thread id {session_id}" not in output.lower()
     ), output
-    assert requests and requests[-1][0] == "/v1/responses", (
-        f"Codex did not resume the zstd rollout\n{output}"
-    )
+    response_request_for_prompt(append_token)
     assert not compressed_history_file.exists(), (
         "Codex left the compressed rollout after append"
     )

